include/exadg/incompressible_navier_stokes/user_interface/parameters.cpp

```diff
@@ -538,14 +537,92 @@ Parameters::check(dealii::ConditionalOStream const & pcout) const
+         GeneralizedNewtonianModel::Carreau ||
+       viscosity_model_data.generalized_newtonian_model_data.generalized_newtonian_model ==
+         GeneralizedNewtonianModel::Cross ||
+       viscosity_model_data.generalized_newtonian_model_data.generalized_newtonian_model ==
+         GeneralizedNewtonianModel::SimplifiedCross)
+    {
+      AssertThrow(std::abs(viscosity_model_data.generalized_newtonian_model_data.kappa - 1.0) <
+                    1e-20,
+                  dealii::ExcMessage("generalized_newtonian_kappa == 1 required for "
+                                     "this GeneralizedNewtonianModel."));
+    }
+
+    if(viscosity_model_data.generalized_newtonian_model_data.generalized_newtonian_model ==
+       GeneralizedNewtonianModel::Carreau)
+    {
+      AssertThrow(std::abs(viscosity_model_data.generalized_newtonian_model_data.a - 2.0) < 1e-20,
+                  dealii::ExcMessage("generalized_newtonian_a == 2 required for"
+                                     "GeneralizedNewtonianModel::Carreau."));
+    }
+
+    if(viscosity_model_data.generalized_newtonian_model_data.generalized_newtonian_model ==
+       GeneralizedNewtonianModel::Cross)
+    {
+      AssertThrow(std::abs(viscosity_model_data.generalized_newtonian_model_data.n - 1.0 +
+                           viscosity_model_data.generalized_newtonian_model_data.a) < 1e-20,
+                  dealii::ExcMessage("generalized_newtonian_n - 1 == generalized_newtonian_a "
+                                     "required for GeneralizedNewtonianModel::Cross."));
+    }
+
+    if(viscosity_model_data.generalized_newtonian_model_data.generalized_newtonian_model ==
+       GeneralizedNewtonianModel::SimplifiedCross)
+    {
+      AssertThrow(std::abs(viscosity_model_data.generalized_newtonian_model_data.a - 1.0) < 1e-20,
+                  dealii::ExcMessage("generalized_newtonian_a == 1 "
+                                     "required for GeneralizedNewtonianModel::SimplifiedCross."));
+      AssertThrow(std::abs(viscosity_model_data.generalized_newtonian_model_data.n) < 1e-20,
+                  dealii::ExcMessage("generalized_newtonian_n == 0 "
+                                     "required for GeneralizedNewtonianModel::SimplifiedCross."));
+    }
+
+    if(viscosity_model_data.generalized_newtonian_model_data.generalized_newtonian_model ==
+       GeneralizedNewtonianModel::PowerLaw)
+    {
+      AssertThrow(std::abs(viscosity) < 1e-20,
+                  dealii::ExcMessage(
+                    "viscosity (= generalized_newtonian_kinematic_viscosity_lower_limit) == 0 "
+                    "required for GeneralizedNewtonianModel::PowerLaw."));
+      AssertThrow(std::abs(viscosity_model_data.generalized_newtonian_model_data.kappa) < 1e-20,
+                  dealii::ExcMessage("generalized_newtonian_kappa == 0 required for "
+                                     "required for GeneralizedNewtonianModel::PowerLaw."));
+    }
```

Comment:
this code might suggest to add a member function `check()` to the struct for the generalized newtonian model data, in order to have the different concerns separated better (single responsibility principle). I admit that this has not been followed strictly in the past (see e.g. turbulence model above), probably because the checks were not that lengthy.